extensions/datastores/kudu/src/main/java/org/locationtech/geowave/datastore/kudu/operations/KuduMetadataReader.java

```diff
@@ -0,0 +1,98 @@
+package org.locationtech.geowave.datastore.kudu.operations;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Streams;
+import org.apache.kudu.Schema;
+import org.apache.kudu.client.*;
+import org.locationtech.geowave.core.store.CloseableIterator;
+import org.locationtech.geowave.core.store.entities.GeoWaveMetadata;
+import org.locationtech.geowave.core.store.operations.MetadataQuery;
+import org.locationtech.geowave.core.store.operations.MetadataReader;
+import org.locationtech.geowave.core.store.operations.MetadataType;
+import org.locationtech.geowave.core.store.operations.RowReader;
+import org.locationtech.geowave.core.store.util.StatisticsRowIterator;
+import org.locationtech.geowave.datastore.kudu.KuduMetadataRow.KuduMetadataField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.*;
+
+public class KuduMetadataReader implements MetadataReader {
+  private final KuduOperations operations;
+  private final MetadataType metadataType;
+  private static final Logger LOGGER = LoggerFactory.getLogger(KuduWriter.class);
+
+
+  public KuduMetadataReader(KuduOperations operations, MetadataType metadataType) {
+    this.operations = operations;
+    this.metadataType = metadataType;
+  }
+
+  @Override
+  public CloseableIterator<GeoWaveMetadata> query(MetadataQuery query) {
+    final String tableName = operations.getMetadataTableName(metadataType);
+    List<RowResultIterator> queryResult = new ArrayList<>();
+    try {
+      KuduTable table = operations.getTable(tableName);
+      Schema schema = table.getSchema();
+      KuduScanner.KuduScannerBuilder scannerBuilder = operations.getScannerBuilder(table);
+      KuduScanner scanner;
+      if (query.hasPrimaryId()) {
+        KuduPredicate primaryPred =
+            KuduPredicate.newComparisonPredicate(
+                schema.getColumn(KuduMetadataField.GW_PRIMARY_ID_KEY.getFieldName()),
+                KuduPredicate.ComparisonOp.EQUAL,
+                query.getPrimaryId());
+        if (query.hasSecondaryId()) {
```

Comment:
We can probably combine the two checks for `query.hasSecondaryId()`. We can change this to:
```
if (query.hasPrimaryId()) {
  // create primaryPred and add to scanner builder
}
if (query.hasSecondaryId()) {
  // create secondaryPred and add to scanner builder
}
scanner = scannerBuilder.build();